tests/extract/test_extract_pipe.py

```diff
@@ -603,6 +603,10 @@ def pass_gen(item, meta):
 
 
 def test_close_on_async_exception() -> None:
+    global close_pipe_got_exit, close_pipe_yielding
+    close_pipe_got_exit = False
+    close_pipe_yielding = False
```

Comment:
globals need to be reset in each test, they will not recover between tests in one file!